### DISCUSSION TITLE
ekg-tags-complete functionality

### DISCUSSION THREAD

**Discussion by monkpearman** [ahyatt/ekg#181] [Ideas]
Some basic built-in ekg-tags completion features would be useful (esp. for beginning ekg users). 
I've added some functions below which do just that. 
Presumably, as the number of tags increase they may need to be filtered, hence I've used CL style keywords in anticipation of that eventual need and to cut down on the cognitive overhead of 8 &optional arguments.

```lisp
(cl-defun ekg-tags-complete (&key prompt collection predicate require-match 
                                  initial-input hist def inherit-input-method)
  "Select an ekg tag from the current `ekg-db' as if by `completing-read'.
Keyword arguments PROMPT, COLLECTION, PREDICATE, REQUIRE-MATCH, INITIAL-INPUT, HIST,
DEF, and INHERIT-INPUT-METHOD are as per `completing-read', which see.
See also `ekg-tags-complete-doc', `ekg-tags'."
  (interactive)
  (completing-read
   (or prompt "Select an ekg tag: ")
   (or collection (ekg-tags))
   predicate
   require-match
   initial-input
   hist
   def
   inherit-input-method))

(defun ekg-tags-complete-doc ()
  "Select a doc/tag from the current `ekg-db' as if by `completing-read'.
See also `ekg-tags-complete'."
  (interactive)
  (ekg-tags-complete :prompt "Select an ekg \"doc/<TAG>\" tag: "
                     :initial-input "doc/"))
```

These functions are included in [pull request 182](https://github.com/ahyatt/ekg/pull/182).

**Comment by ahyatt** [comment]
Thanks for contributing that!  I'll take a look at the pull request, and comment on that.  I'm curious how you see these being used in the future.
